website/source/docs/internals/architecture.html.md

```diff
@@ -87,10 +87,10 @@ availability in the case of failure and performance, as consensus gets progressi
 slower as more machines are added. However, there is no limit to the number of clients,
 and they can easily scale into the thousands or tens of thousands.
 
-All the nodes that are in a datacenter participate in a [gossip protocol](/docs/internals/gossip.html).
-This means there is a gossip pool that contains all the nodes for a given datacenter. This serves
+All the agents that are in a datacenter participate in a [gossip protocol](/docs/internals/gossip.html).
+This means there is a gossip pool that contains all the agents for a given datacenter. This serves
 a few purposes: first, there is no need to configure clients with the addresses of servers;
-discovery is done automatically. Second, the work of detecting node failures
+discovery is done automatically. Second, the work of detecting agent failures
```

Comment:
This is the only one I'm not sure about. Technically it's true but the real value here is detecting "node" failures more generally. I.e. you care that the whole VM is down and hence your workloads on it are not reachable rather than caring if the Consul agent specifically is up and running. You could just revert to "node" in this one case I think clearly enough or describe that more elaborately but I think it's worth pointing out that this is how Consul can automatically route around failures that affect your actual services rather than just internal consul agent failures.

Reply:
That was definitely on my mind when I updated it from node to agent. However, I was wondering if there are cases when your agent (Consul client) process would fail, but not the actual node. I agree that failure detection is definitely an infrastructure benefit. I 'll try to add another sentence, if I can make it fit.